libraries/abstractions/pkcs11/utest/iot_pkcs11_mbedtls_utest.c

```diff
@@ -171,112 +299,540 @@ void test_pkcs11_C_InitializeInitTwice( void )
+    CK_BBOOL xTokenPresent = CK_TRUE;
 
-    xQueueCreateMutex_IgnoreAndReturn( ( SemaphoreHandle_t ) &xResult );
-    CRYPTO_Init_Ignore();
-    mbedtls_entropy_init_Ignore();
-    mbedtls_ctr_drbg_init_Ignore();
-    mbedtls_ctr_drbg_seed_IgnoreAndReturn( 0 );
-    xResult = C_Initialize( NULL );
+    xResult = prvInitializePkcs11( ( SemaphoreHandle_t ) &xResult );
+    TEST_ASSERT_EQUAL( CKR_OK, xResult );
 
+    xResult = C_GetSlotList( xTokenPresent, pxSlotId, &xSlotCount );
     TEST_ASSERT_EQUAL( CKR_OK, xResult );
+    TEST_ASSERT_EQUAL( 1, xSlotCount );
 
-    mbedtls_entropy_free_CMockIgnore();
-    mbedtls_ctr_drbg_free_CMockIgnore();
-    vQueueDelete_CMockIgnore();
-    xResult = C_Finalize( NULL );
+    pxSlotId = pvPkcs11MallocCb( sizeof( CK_SLOT_ID ) * xSlotCount, 1 );
+    memset( pxSlotId, 0, sizeof( CK_SLOT_ID ) * 1 );
+
+    xResult = C_GetSlotList( xTokenPresent, pxSlotId, &xSlotCount );
 
     TEST_ASSERT_EQUAL( CKR_OK, xResult );
+    TEST_ASSERT_EQUAL( 1, xSlotCount );
+    TEST_ASSERT_EQUAL( 1, pxSlotId[ 0 ] );
+
+    vPkcs11FreeCb( pxSlotId, 1 );
+
+    xResult = prvUninitializePkcs11();
+    TEST_ASSERT_EQUAL( CKR_OK, xResult );
 }
 
 /*!
- * @brief C_Finalize initialize was not already called.
+ * @brief C_GetSlotList NULL SlotList.
  *
  */
-void test_pkcs11_C_FinalizeUninitialized( void )
+void test_pkcs11_C_GetSlotListNullSlot( void )
 {
     CK_RV xResult = CKR_OK;
+    CK_ULONG xSlotCount = 0;
+    CK_BBOOL xTokenPresent = CK_TRUE;
 
-    mbedtls_entropy_free_CMockIgnore();
-    mbedtls_ctr_drbg_free_CMockIgnore();
-    vQueueDelete_CMockIgnore();
-    xResult = C_Finalize( NULL );
+    xResult = prvInitializePkcs11( ( SemaphoreHandle_t ) &xResult );
+    TEST_ASSERT_EQUAL( CKR_OK, xResult );
+
+    xResult = C_GetSlotList( xTokenPresent, NULL, &xSlotCount );
+    TEST_ASSERT_EQUAL( CKR_OK, xResult );
+
+    TEST_ASSERT_EQUAL( CKR_OK, xResult );
+    TEST_ASSERT_EQUAL( 1, xSlotCount );
+
+    xResult = prvUninitializePkcs11();
+    TEST_ASSERT_EQUAL( CKR_OK, xResult );
+}
+
+/*!
+ * @brief C_GetSlotList NULL SlotCount.
+ *
+ */
+void test_pkcs11_C_GetSlotListNullCount( void )
+{
+    CK_RV xResult = CKR_OK;
+    CK_BBOOL xTokenPresent = CK_TRUE;
+
+    xResult = prvInitializePkcs11( ( SemaphoreHandle_t ) &xResult );
+    TEST_ASSERT_EQUAL( CKR_OK, xResult );
+
+    xResult = C_GetSlotList( xTokenPresent, NULL, NULL );
+    TEST_ASSERT_EQUAL( CKR_ARGUMENTS_BAD, xResult );
+
+    xResult = prvUninitializePkcs11();
+    TEST_ASSERT_EQUAL( CKR_OK, xResult );
+}
+
+/*!
+ * @brief C_GetSlotList 0 object count, valid slotlist.
+ *
+ */
+void test_pkcs11_C_GetSlotListBadObjCount( void )
+{
+    CK_RV xResult = CKR_OK;
+    CK_SLOT_ID_PTR pxSlotId = NULL;
+    CK_ULONG xSlotCount = 0;
+    CK_BBOOL xTokenPresent = CK_TRUE;
+
+    xResult = prvInitializePkcs11( ( SemaphoreHandle_t ) &xResult );
+    TEST_ASSERT_EQUAL( CKR_OK, xResult );
+
+    pxSlotId = pvPkcs11MallocCb( sizeof( CK_SLOT_ID ) * 1, 1 );
```

Comment:
If we're going to assume that there's only one slot, why use malloc?

Reply:
Fixed this